### PR TITLE
Tedlee/enable skill auth

### PR DESCRIPTION
**Pull request by lzc850612**
## Description

Enable ADAL fashion auth for skill invocation. 

ParentBot: acquire JWT token from AAD and put it in the bearer header
SkillBot: verify JWT token, and optionally handle appid to see if it's a ParentBot that earlier added the skill

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the appropriate unit tests
- [ ] I have tested any new/updated dialogs using Speech in the emulator to ensure the [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) property is set to enable a high quality speech-first experience and the appropriate [InputHints](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) are set correctly. 
- [ ] I have updated related documentation

- [ ] A duplicate issue is filed to track future  work.

- [ ] All languages have been updated
- [ ] You have tested deployment with your new models

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/skills/emailskill/emailskill/Startup.cs [hunk 1]
```diff
@@ -4,7 +4,12 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
 using System.Linq;
+<<<<<<< HEAD
```
Comment (DingmaomaoBJTU) [inline solutions/Virtual-Assistant/src/csharp/skills/emailskill/emailskill/Startup.cs]: Resolve this?

File: solutions/Virtual-Assistant/src/csharp/skills/emailskill/emailskill/Startup.cs [hunk 2]
```diff
@@ -125,6 +133,20 @@ public void ConfigureServices(IServiceCollection services)
...
             services.AddSingleton(endpointService);
 
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+            .AddJwtBearer(options =>
+            {
+                options.Authority = "https://login.microsoftonline.com/microsoft.com";
+                options.Audience = endpointService.AppId;
+                options.TokenValidationParameters = new TokenValidationParameters
+                {
+                    ValidIssuer = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/v2.0",
```
Comment (DingmaomaoBJTU) [inline solutions/Virtual-Assistant/src/csharp/skills/emailskill/emailskill/Startup.cs:138]: Just curious, what is this?

File: solutions/Virtual-Assistant/src/csharp/experimental/skills/restaurantbooking/SkillWhitelist.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using Microsoft.Bot.Builder.Skills.Auth;
+
+namespace RestaurantBooking
+{
+    public class SkillWhitelist : ISkillWhitelist
+    {
+        private readonly List<string> _whiteList = new List<string>
+        {
+            "7cc1ea17-ceec-42fe-b056-45c25884d7b7"
```
Comment (lauren-mills) [inline solutions/Virtual-Assistant/src/csharp/experimental/skills/restaurantbooking/SkillWhitelist.cs:10]: Should we drive this based on config?

Comment (lzc850612) [reply solutions/Virtual-Assistant/src/csharp/experimental/skills/restaurantbooking/SkillWhitelist.cs:10]: The whitelist actually in long term should be off of state (minus the turnContext). so we'll use this approach for now.

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**